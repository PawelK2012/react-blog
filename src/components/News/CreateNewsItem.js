import React, { Component } from 'react'

class CreateNewsItem extends Component {
    render() {
        return (
          <h1>CreateNewsItem</h1>
        )
      }
}

export default CreateNewsItem