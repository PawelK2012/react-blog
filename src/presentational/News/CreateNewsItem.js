import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'material-ui/TextField'
import Toggle from 'material-ui/Toggle'
import RaisedButton from 'material-ui/RaisedButton'

const styles = {
  block: {
    maxWidth: 150,
  },
  toggle: {
    marginBottom: 16,
  },
  textField: {
    marginBottom: '15px'
  },
  button: {
    marginLeft: 12,
    float: 'right',
  }
};

export const CreateNewsItem = ({ title, subTitle, text, owner, img, featured, onSave }) => {
  let _title, _subTitle, _text, _owner, _img, _featured

  const save = (e) => {
    e.preventDefaults;
    onSave({
      title: _title.getValue(),
      subTitle: _subTitle.getValue(),
      text: _text.getValue(),
      owner: owner,
      img: _img.getValue(),
      featured: _featured.state.switched
    })
  }

  return (
    <div className="create-news-container">
      <p>Create news</p>
      <form onSubmit={save} className="create-news-form">
        <TextField
          id={'title'}
          hintText="Title"
          ref={input => _title = input}
          fullWidth={true}
          style={styles.textField}
        />
        <TextField
          hintText="Sub Title"
          ref={input => _subTitle = input}
          fullWidth={true}
          style={styles.textField}
        />
        <TextField
          hintText="News Field"
          ref={input => _text = input}
          multiLine={true}
          rows={4}
          fullWidth={true}
          style={styles.textField}
        />
        <TextField
          hintText="URL to your image"
          ref={input => _img = input}
          fullWidth={true}
          style={styles.textField}
          type={'url'}
        />
        <div style={styles.block}>
          <Toggle
            label="Featured item"
            ref={input => _featured = input}
            style={styles.toggle}
            defaultToggled={false}
          />
        </div>
        <RaisedButton label="Save" onClick={save} secondary={true} style={styles.button} />
        <RaisedButton label="Clear" primary={true} style={styles.button} />
      </form>
    </div>
  )
}

CreateNewsItem.defaultProps = {
  title: 'Some title',
  subTitle: 'Some subtitle',
  text: 'News content',
  owner: 'Paa',
  img: 'http://dailypost.ng/wp-content/uploads/2017/10/lew.jpg',
  featured: false
}

CreateNewsItem.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  featured: PropTypes.bool.isRequired
}

export default CreateNewsItem
