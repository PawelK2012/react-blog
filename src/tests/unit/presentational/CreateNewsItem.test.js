import React from 'react';
import { shallow } from 'enzyme';
import CreateNewsItem from '../../../presentational/News/CreateNewsItem';

describe('CreateNewsItem Component:', () => { 
    test('renders correctly', () => {
      const wrapper = shallow(<CreateNewsItem />);
      expect(wrapper.html()).toMatchSnapshot();
    });
});