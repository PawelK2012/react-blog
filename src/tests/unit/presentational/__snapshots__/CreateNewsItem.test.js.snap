// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewsItem Component: renders correctly 1`] = `
"<div class=\\"create-news-container\\"><p>Create news</p><form class=\\"create-news-form\\"><div style=\\"font-size:16px;line-height:24px;width:100%;height:48px;display:inline-block;position:relative;background-color:transparent;font-family:Roboto, sans-serif;transition:height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;cursor:auto;margin-bottom:15px;mui-prepared:\\"><div style=\\"position:absolute;opacity:1;color:rgba(0, 0, 0, 0.3);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;bottom:12px;mui-prepared:\\">Title</div><input type=\\"text\\" style=\\"appearance:textfield;padding:0;position:relative;width:100%;border:none;outline:none;background-color:rgba(0,0,0,0);color:rgba(0, 0, 0, 0.87);cursor:inherit;font:inherit;-webkit-opacity:1;-webkit-tap-highlight-color:rgba(0,0,0,0);height:100%;mui-prepared:\\" id=\\"title\\"/><div><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:1px;border-color:#e0e0e0;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;mui-prepared:\\"/><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:2px;border-color:#00bcd4;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;transform:scaleX(0);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:\\"/></div></div><div style=\\"font-size:16px;line-height:24px;width:100%;height:48px;display:inline-block;position:relative;background-color:transparent;font-family:Roboto, sans-serif;transition:height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;cursor:auto;margin-bottom:15px;mui-prepared:\\"><div style=\\"position:absolute;opacity:1;color:rgba(0, 0, 0, 0.3);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;bottom:12px;mui-prepared:\\">Sub Title</div><input type=\\"text\\" style=\\"appearance:textfield;padding:0;position:relative;width:100%;border:none;outline:none;background-color:rgba(0,0,0,0);color:rgba(0, 0, 0, 0.87);cursor:inherit;font:inherit;-webkit-opacity:1;-webkit-tap-highlight-color:rgba(0,0,0,0);height:100%;mui-prepared:\\" id=\\"undefined-SubTitle-undefined-52419\\"/><div><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:1px;border-color:#e0e0e0;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;mui-prepared:\\"/><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:2px;border-color:#00bcd4;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;transform:scaleX(0);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:\\"/></div></div><div style=\\"font-size:16px;line-height:24px;width:100%;height:120px;display:inline-block;position:relative;background-color:transparent;font-family:Roboto, sans-serif;transition:height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;cursor:auto;margin-bottom:15px;mui-prepared:\\"><div style=\\"position:absolute;opacity:1;color:rgba(0, 0, 0, 0.3);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;bottom:12px;mui-prepared:\\">News Field</div><div style=\\"position:relative;padding:0;width:100%;border:none;outline:none;background-color:rgba(0,0,0,0);color:rgba(0, 0, 0, 0.87);cursor:inherit;font:inherit;-webkit-opacity:1;-webkit-tap-highlight-color:rgba(0,0,0,0);height:100%;mui-prepared:\\"><textarea style=\\"height:auto;width:100%;resize:none;font:inherit;padding:0;cursor:inherit;position:absolute;border:none;outline:none;background-color:rgba(0,0,0,0);color:rgba(0, 0, 0, 0.87);-webkit-opacity:1;-webkit-tap-highlight-color:rgba(0,0,0,0);margin-top:12px;margin-bottom:-12px;box-sizing:border-box;appearance:textfield;overflow:hidden;visibility:hidden;mui-prepared:\\" tabindex=\\"-1\\" rows=\\"4\\" readonly=\\"\\"></textarea><textarea id=\\"undefined-NewsField-undefined-41254\\" rows=\\"4\\" style=\\"height:96px;width:100%;resize:none;font:inherit;padding:0;cursor:inherit;position:relative;border:none;outline:none;background-color:rgba(0,0,0,0);color:rgba(0, 0, 0, 0.87);-webkit-opacity:1;-webkit-tap-highlight-color:rgba(0,0,0,0);margin-top:12px;margin-bottom:-12px;box-sizing:border-box;appearance:textfield;mui-prepared:\\"></textarea></div><div><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:1px;border-color:#e0e0e0;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;mui-prepared:\\"/><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:2px;border-color:#00bcd4;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;transform:scaleX(0);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:\\"/></div></div><div style=\\"font-size:16px;line-height:24px;width:100%;height:48px;display:inline-block;position:relative;background-color:transparent;font-family:Roboto, sans-serif;transition:height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;cursor:auto;margin-bottom:15px;mui-prepared:\\"><div style=\\"position:absolute;opacity:1;color:rgba(0, 0, 0, 0.3);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;bottom:12px;mui-prepared:\\">URL to your image</div><input type=\\"url\\" style=\\"appearance:textfield;padding:0;position:relative;width:100%;border:none;outline:none;background-color:rgba(0,0,0,0);color:rgba(0, 0, 0, 0.87);cursor:inherit;font:inherit;-webkit-opacity:1;-webkit-tap-highlight-color:rgba(0,0,0,0);height:100%;mui-prepared:\\" id=\\"undefined-URLtoyourimage-undefined-23947\\"/><div><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:1px;border-color:#e0e0e0;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;mui-prepared:\\"/><hr aria-hidden=\\"true\\" style=\\"border-top:none;border-left:none;border-right:none;border-bottom-style:solid;border-bottom-width:2px;border-color:#00bcd4;bottom:8px;box-sizing:content-box;margin:0;position:absolute;width:100%;transform:scaleX(0);transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;mui-prepared:\\"/></div></div><div style=\\"max-width:150px\\"><div style=\\"cursor:pointer;position:relative;overflow:visible;display:table;height:auto;width:100%;margin-bottom:16px;mui-prepared:\\"><input type=\\"checkbox\\" style=\\"position:absolute;cursor:inherit;pointer-events:all;opacity:0;width:100%;height:100%;z-index:2;left:0;box-sizing:border-box;padding:0;margin:0;mui-prepared:\\"/><div style=\\"display:flex;width:100%;height:100%\\"><label style=\\"float:left;position:relative;display:block;width:calc(100% - 46px);line-height:24px;color:rgba(0, 0, 0, 0.87);font-family:Roboto, sans-serif;mui-prepared:\\">Featured item</label><div style=\\"transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;float:left;position:relative;display:block;flex-shrink:0;width:36px;margin-right:0;margin-left:8px;padding:4px 0px 6px 2px;mui-prepared:\\"><div style=\\"transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;width:100%;height:14px;border-radius:30px;background-color:#bdbdbd;mui-prepared:\\"></div><div style=\\"color:rgba(0, 0, 0, 0.87);background-color:#f5f5f5;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;box-sizing:border-box;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0,0,0,0);box-shadow:0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12);border-radius:50%;position:absolute;top:1px;left:0;width:20px;height:20px;line-height:24px;mui-prepared:\\"> <div></div><div style=\\"position:absolute;height:100%;width:100%;top:0;left:0;mui-prepared:\\"></div> </div></div></div></div></div><div style=\\"color:rgba(0, 0, 0, 0.87);background-color:#ffffff;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;box-sizing:border-box;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0,0,0,0);box-shadow:0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12);border-radius:2px;display:inline-block;min-width:88px;margin-left:12px;float:right;mui-prepared:\\"><button style=\\"border:10px;box-sizing:border-box;display:inline-block;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0, 0, 0, 0);cursor:pointer;text-decoration:none;margin:0;padding:0;outline:none;font-size:inherit;font-weight:inherit;position:relative;height:36px;line-height:36px;width:100%;border-radius:2px;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;background-color:#ff4081;text-align:center;mui-prepared:\\" tabindex=\\"0\\" type=\\"button\\"><div><div style=\\"height:36px;border-radius:2px;background-color:;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;top:0;mui-prepared:\\"><span style=\\"position:relative;opacity:1;font-size:14px;letter-spacing:0;text-transform:uppercase;font-weight:500;margin:0;user-select:none;padding-left:16px;padding-right:16px;color:#ffffff;mui-prepared:\\">Save</span></div></div></button></div><div style=\\"color:rgba(0, 0, 0, 0.87);background-color:#ffffff;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;box-sizing:border-box;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0,0,0,0);box-shadow:0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12);border-radius:2px;display:inline-block;min-width:88px;margin-left:12px;float:right;mui-prepared:\\"><button style=\\"border:10px;box-sizing:border-box;display:inline-block;font-family:Roboto, sans-serif;-webkit-tap-highlight-color:rgba(0, 0, 0, 0);cursor:pointer;text-decoration:none;margin:0;padding:0;outline:none;font-size:inherit;font-weight:inherit;position:relative;height:36px;line-height:36px;width:100%;border-radius:2px;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;background-color:#00bcd4;text-align:center;mui-prepared:\\" tabindex=\\"0\\" type=\\"button\\"><div><div style=\\"height:36px;border-radius:2px;background-color:;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;top:0;mui-prepared:\\"><span style=\\"position:relative;opacity:1;font-size:14px;letter-spacing:0;text-transform:uppercase;font-weight:500;margin:0;user-select:none;padding-left:16px;padding-right:16px;color:#ffffff;mui-prepared:\\">Clear</span></div></div></button></div></form></div>"
`;
